Explorer/Assets/DCL/Multiplayer/Movement/Systems/SDKAvatarShapesMotionSystem.cs

```diff
@@ -0,0 +1,108 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.Character.Components;
+using DCL.CharacterMotion.Animation;
+using DCL.CharacterMotion.Components;
+using DCL.Diagnostics;
+using DCL.Profiles;
+using ECS.Abstract;
+using UnityEngine;
+
+namespace DCL.Multiplayer.Movement.Systems
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [LogCategory(ReportCategory.AVATAR)]
+    public partial class SDKAvatarShapesMotionSystem : BaseUnityLoopSystem
+    {
+        private const float ROTATION_SPEED = 5f;
+        private const float RUN_SPEED_THRESHOLD = 9.5f;
+        private const float JOG_SPEED_THRESHOLD = 4f;
+
+        private SDKAvatarShapesMotionSystem(World world) : base(world) { }
+
+        protected override void Update(float t)
+        {
+            UpdateMotionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(Profile))]
+        private void UpdateMotion(
+            in IAvatarView view,
+            ref CharacterTransform characterTransformComponent,
+            ref CharacterLastPosition characterLastPositionComponent,
+            ref CharacterAnimationComponent animationComponent)
+        {
+            UpdateRotation(characterTransformComponent, ref characterLastPositionComponent);
+            UpdateAnimations(view, characterTransformComponent, ref characterLastPositionComponent, ref animationComponent);
+            UpdateLastPosition(ref characterTransformComponent, ref characterLastPositionComponent);
+        }
+
+        private static void UpdateRotation(
+            CharacterTransform characterTransformComp,
+            ref CharacterLastPosition characterLastPositionComp)
+        {
+            var currentPosition = characterTransformComp.Transform.position;
+            var oldPosition = characterLastPositionComp.LastPosition;
+
+            if (oldPosition == currentPosition)
+                return;
+
+            Vector3 newDirection = currentPosition - oldPosition;
+            newDirection.y = 0;
+
+            if (newDirection != Vector3.zero)
+            {
+                Vector3 currentForward = characterTransformComp.Transform.forward;
+                Vector3 targetForward = newDirection.normalized;
```

Comment:
it is just a rotation, so I would optimize and just drop normalization of newDirection. it is not needed since it keeps same  vector direction but  normalize its magnitude to 1 (and you don't care about it when working with rotations). Just remove that like

Reply:
As we are using this:
```
if (Vector3.Dot(currentForward, targetForward) < 0.999f)
```

I want to compare both vectors normalized.